azurerm/internal/services/appplatform/resource_arm_spring_cloud_config_server.go

```diff
@@ -0,0 +1,530 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSpringCloudConfigServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSpringCloudConfigServerCreateUpdate,
+		Read:   resourceArmSpringCloudConfigServerRead,
+		Update: resourceArmSpringCloudConfigServerCreateUpdate,
+		Delete: resourceArmSpringCloudConfigServerDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.SpringCloudID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"spring_cloud_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SpringCloudID,
+			},
+
+			"git_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"uri": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: ValidateConfigServerURI,
+						},
+
+						"label": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"search_paths": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+
+						"http_basic_auth": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"username": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"password": {
+										Type:      schema.TypeString,
+										Required:  true,
+										Sensitive: true,
+									},
+								},
+							},
+							ConflictsWith: []string{"git_settings.0.ssh_auth"},
+						},
+
+						"ssh_auth": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"private_key": {
+										Type:      schema.TypeString,
+										Required:  true,
+										Sensitive: true,
+									},
+
+									"host_key": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										Sensitive: true,
+									},
+
+									"host_key_algorithm": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+
+									"strict_host_key_checking_enabled": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+								},
+							},
+							ConflictsWith: []string{"git_settings.0.http_basic_auth"},
+						},
+
+						"repositories": {
+							Type:     schema.TypeList,
```

Comment:
because we should look up some sensitive data from the nested block, and the maxItem is not 1. I think use List is convenient to loop up data by index